### PR TITLE
Const changes to support new VFP Planner API refactor.

### PR DESCRIPTION
**Pull request by sniyaz**
This PR has `const` changes to support the upcoming VFP refactor onto the new Planner API. https://github.com/personalrobotics/aikido/pull/426 is currently blocking on this PR. Once this PR has been merged, I will rebase https://github.com/personalrobotics/aikido/pull/426, and we can tackle that PR.

Because the new `Problem` derivate classes from the new Planner API (https://github.com/personalrobotics/aikido/pull/314) make most members `const`, this requires shoving a bunch of `const` variables through AIKIDO (we probably don't want to cast the `const`-ness away, since this would break a promise to the user).

These changes are necessary to make `planToEndEffectorOffset` possible with the new Planner API. `planToEndEffectorPose` may require more changes- my plan is to get `planToEndEffectorOffset` refactored first, then tackle `planToEndEffectorPose`.

**Summary of changes**

1) Const all the things.

2) Update old `planToEndEffectorOffset`  with the signature it needs to have in https://github.com/personalrobotics/aikido/pull/426.

***

**Before creating a pull request**

- [x] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [x] Set version target by selecting a milestone on the right side
- [x] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

### CODE REVIEW

File: include/aikido/constraint/dart/FramePairDifferentiable.hpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ class FramePairDifferentiable : public Differentiable
   std::vector<ConstraintType> getConstraintTypes() const override;
 
   // Documentation inherited.
-  statespace::StateSpacePtr getStateSpace() const override;
+  statespace::ConstStateSpacePtr getStateSpace() const override;
 
 private:
   statespace::dart::MetaSkeletonStateSpacePtr mMetaSkeletonStateSpace;
```
[the remark this answers is not in the document]
Comment (sniyaz) [reply include/aikido/constraint/dart/FramePairDifferentiable.hpp]: It looks like we need the return type to match for the override to work.

File: include/aikido/constraint/dart/FrameDifferentiable.hpp [hunk 2]
```diff
@@ -62,7 +62,7 @@ class FrameDifferentiable : public Differentiable
   std::vector<ConstraintType> getConstraintTypes() const override;
 
   // Documentation inherited.
-  statespace::StateSpacePtr getStateSpace() const override;
+  statespace::ConstStateSpacePtr getStateSpace() const override;
 
 private:
   statespace::dart::MetaSkeletonStateSpacePtr mMetaSkeletonStateSpace;
```
[the remark this answers is not in the document]
Comment (sniyaz) [reply include/aikido/constraint/dart/FrameDifferentiable.hpp]: It looks like we need the return type to match for the override to work.

File: include/aikido/constraint/dart/FramePairDifferentiable.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (brianhou) [reply include/aikido/constraint/dart/FramePairDifferentiable.hpp]: Sorry, this comment was about `mMetaSkeletonStateSpace`.

File: include/aikido/constraint/dart/FrameDifferentiable.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (brianhou) [reply include/aikido/constraint/dart/FrameDifferentiable.hpp]: Sorry, this comment was about `mMetaSkeletonStateSpace`.

File: include/aikido/constraint/dart/FrameTestable.hpp [hunk 3]
```diff
@@ -45,14 +45,14 @@ class FrameTestable : public Testable
   std::unique_ptr<TestableOutcome> createOutcome() const override;
 
   // Documentation inherited
-  std::shared_ptr<statespace::StateSpace> getStateSpace() const override;
+  statespace::ConstStateSpacePtr getStateSpace() const override;
 
 private:
   statespace::dart::MetaSkeletonStateSpacePtr mMetaSkeletonStateSpace;
```
[the remark this answers is not in the document]
Comment (sniyaz) [reply include/aikido/constraint/dart/FrameTestable.hpp]: It looks like this breaks the override.

File: include/aikido/constraint/dart/InverseKinematicsSampleable.hpp [hunk 4]
```diff
@@ -41,7 +41,7 @@ class InverseKinematicsSampleable : public Sampleable
   virtual ~InverseKinematicsSampleable() = default;
 
   // Documentation inherited.
-  statespace::StateSpacePtr getStateSpace() const override;
+  statespace::ConstStateSpacePtr getStateSpace() const override;
```
[the remark this answers is not in the document]
Comment (sniyaz) [reply include/aikido/constraint/dart/InverseKinematicsSampleable.hpp]: It looks like this breaks the override.

File: include/aikido/constraint/dart/FramePairDifferentiable.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sniyaz) [reply include/aikido/constraint/dart/FramePairDifferentiable.hpp]: Derp derp, you're right! 😅

**Comment by sniyaz** [comment]
Responded to @brianhou's comments- @jslee02, could you take a look when you have the chance?

**Pull request merged**